src/sql/workbench/contrib/notebook/test/electron-browser/notebookFindModel.test.ts

```diff
@@ -184,11 +184,94 @@ suite('Notebook Find Model', function (): void {
 		await initNotebookModel(codeContent);
 		//initialize find
 		let notebookFindModel = new NotebookFindModel(model);
-		await notebookFindModel.find('x', max_find_count);
+		await notebookFindModel.find('x', false, false, max_find_count);
 
 		assert.equal(notebookFindModel.findMatches.length, 3, 'Find failed');
 	});
 
+	test('Should find results correctly with & without matching case selection', async function (): Promise<void> {
+		//initialize find
+		let notebookFindModel = new NotebookFindModel(model);
+		await notebookFindModel.find('insert', false, false, max_find_count);
+
+		assert(notebookFindModel.findMatches, 'Find in notebook failed.');
+		assert.equal(notebookFindModel.findMatches.length, 3, 'Find couldnt find all occurances');
+
+		await notebookFindModel.find('insert', true, false, max_find_count);
+		assert.equal(notebookFindModel.findMatches.length, 2, 'Find failed to apply match case while searching');
+
+	});
+
+	test('Should find results with matching whole word in the notebook', async function (): Promise<void> {
+		//initialize find
+		let notebookFindModel = new NotebookFindModel(model);
+
+		await notebookFindModel.find('insert', true, true, max_find_count);
+		assert.equal(notebookFindModel.findMatches.length, 1, 'Find failed to apply whole word filter while searching');
+
+	});
+
+	test('Should find special characters in the search term without problems', async function (): Promise<void> {
+		let codeContent: nb.INotebookContents = {
+			cells: [{
+				cell_type: CellTypes.Code,
+				source: ['import x', 'x.init()', '//am just adding a bunch of {special} <characters> !!!!$}'],
+				metadata: { language: 'python' },
+				execution_count: 1
+			}],
+			metadata: {
+				kernelspec: {
+					name: 'python',
+					language: 'python'
+				}
+			},
+			nbformat: 4,
+			nbformat_minor: 5
+		};
+		await initNotebookModel(codeContent);
+		//initialize find
+		let notebookFindModel = new NotebookFindModel(model);
+		// test for string with special character
+		await notebookFindModel.find('{special}', false, false, max_find_count);
+		assert.equal(notebookFindModel.findMatches.length, 1, 'Find failed for search term with spcl character');
+		// test for only special character !!
+		await notebookFindModel.find('!!', false, false, max_find_count);
```

Comment:
It would be best to also test these with match case (even if that should end with the same result) and match word as well

Reply:
done